Refactor: Remove `check_structure()` from `UnitCell`.

sunliang98 opened the pull request:
Remove `check_structure()` from `UnitCell`, as the first step of the refactor of `UnitCell`, and update corresponding unit test.

However, a point of discussion persists: In line 1408 of `unitcell.cpp`, the element type `symbol1` is assigned by `std::string symbol1 = this->atoms[it1].ncpp.psd;`, where `psd` is read from pseudopotential, but `atom[it1].label` holds the  element type read from `STRU` and is unrelated to the pseudopotential information. I have changed `atoms[it1].ncpp.psd` to `atoms[it1].label` in new code, but I am curious as to whether there might be any particular advantage behind employing `atoms[it1].ncpp.psd` over `atoms[it1].label`.


### Reminder
- [x] Have you linked an issue with this pull request?
- [x] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #4498 

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

sunliang98 committed:
<Refactor> Remove check_structure() from UnitCell, as the first step of the refactor of UnitCell.

sunliang98 committed:
<Test> Update Makefile

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

mohanchen commented on source/Makefile.Objects:
```diff
@@ -172,6 +172,7 @@ OBJS_CELL=atom_pseudo.o\
     setup_nonlocal.o\
     klist.o\
     cell_index.o\
+    check_structure.o\
```
change another name？

sunliang98 replied:
OK

sunliang98 committed:
Refactor: Rename `check_structure` to `check_atomic_stru`.
Convert labels (`ucell.atoms[it1].label`) like "Fe1" to "Fe" before checking the atomic structure.

sunliang98 committed:
Merge branch 'cell' of https://github.com/sunliang98/abacus-develop into cell

sunliang98 committed:
refactor: remove "check_structure" module and update dependencies

sunliang98 committed:
Merge branch 'develop' into cell

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

mohanchen committed:
Merge branch 'develop' into cell

mohanchen commented on source/module_cell/unitcell.h:
```diff
@@ -307,7 +307,7 @@ class UnitCell
                const std::string& fixed_axes_in);
 
-    void check_structure(double factor);
+    void check_atomic_stru(double factor);
```
can we delete the 'check_atomic_stru' function?

sunliang98 committed:
Delete check_atomic_stru() in unitcell.h

sunliang98 committed:
Merge branch 'develop' into cell

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes
